docs/rfcs/1005-realm-key-rotation.md

```diff
@@ -0,0 +1,821 @@
+        },
+        {
+            // Empty message encrypted with the key, this is used to ensure the
+            // the key retrieved in the keys-bundle hasn't been tempered (given
+            // each key rotation author generates a new keys-bundle from the
+            // previous bundle and his one additional key).
+            //
+            // Note since the message is empty, we are doing here AEAD-without-data
+            // given we only care of the MAC part.
+            "name": "key_canary",
+            "type": "Bytes"
+        }
+    ]
+}
+```
+
+When processing this certificate (in addition to regular access checks), the
+client must make sure the `key_index` field strictly follow the value of the
+previous key rotation certificate for this realm. (i.e. we reject the
+certificate if there is a gap in key indexes)
+
+`SecretKeyAlgorithm` has one possible values: `XSALSA20-POLY1305`.
+`HashAlgorithm` has one possible value: `SHA256`.
+
+> **TODO:**
+>
+> The fields `encryption_algorithm` & `hash_algorithm` allows for future evolution.
+> The idea would be to switch to  Chacha20-Poly1305 given it is a more scrutinized
+> algorithm with audited implementations.
+>
+> Another benefit is to be able to easily upgrade those algorithm in the future in
+> case of exploit in them.
+
+### 4.4 New document `RealmKeysBundle`
+
+```json5
+{
+    "label": "RealmKeysBundle",
+    "type": "realm_keys_bundle",
+    "other_fields": [
+        {
+            "name": "author",
+            "type": "DeviceID",
+        },
+        {
+            // Timestamp must be equal to the one in the `RealmRoleCertificate`
+            "name": "timestamp",
+            "type": "DateTime"
+        },
+        {
+            "name": "realm",
+            "type": "VlobID"
+        },
+        {
+            "name": "keys",
+            // Keys ordered by index
+            "type": "List<Bytes>"
+        }
+    ]
+}
+```
+
+### 4.5 New document `RealmKeysBundleAccess`
+
+```json5
+{
+    "label": "RealmKeysBundleAccess",
+    "type": "realm_keys_bundle_access",
+    "other_fields": [
+        {
+            "name": "author",
+            "type": "DeviceID"
+        },
+        {
+            // Timestamp must be equal to the one in the `RealmRoleCertificate`
+            // or `RealmKeyRotationCertificate` (depending if the access has been
+            // given before or after last key rotation)
+            "name": "timestamp",
+            "type": "DateTime"
+        },
+        {
+            "name": "realm",
+            "type": "VlobID"
+        },
+        {
+            "name": "key_index",
+            "type": "IndexInt"
+        },
+        {
+            "name": "keys_bundle_key",
+            "type": "SecretKey"
+        },
+        {
+            // Only present when the `RealmKeysBundleAccess` is generated for use with
+            // `realm_share` command (and not `realm_rotate_key` given in this case the
+            // user already )
```

Comment:
already what ?

Reply:
![image](https://github.com/Scille/parsec-cloud/assets/3187637/06c632e2-fa15-4e46-a2a0-7362c20abbe3)

Reply:
this is outdated stuff and will be remove now that we settled on the go light approach for `KeyBundleAccess`